LIBITD-2552. Added Group List UI View

aatishgupta25 opened the pull request:
Made a new template ui/groupListView.html and added a URL path to this view and added a view for listing the groups.

https://umd-dit.atlassian.net/browse/LIBITD-2552

Commit:
LIBITD-2552. Added Group List UI View

Made a new template ui/groupListView.html and added a
URL path to this view and added a view for listing the
groups.

https://umd-dit.atlassian.net/browse/LIBITD-2552

peichman-umd commented on src/ipmanager/ui/views.py:
```diff
@@ -1,8 +1,16 @@
+ model = Group
+ template_name = 'ui/groupListView.html'
+
+ def get_queryset(self):
```
Since the default behavior of the `ListView` is to get the list of all objects from the model, you do not need this explicit `get_queryset()` method.

peichman-umd commented on src/ipmanager/ui/views.py:
```diff
@@ -1,8 +1,16 @@
+
+class GroupListView(ListView):
+ model = Group
+ template_name = 'ui/groupListView.html'
```
Python convention is to use "snake case" for template names (like variables and functions), so this would be `ui/group_list.html`

peichman-umd reviewed (changes requested):
Looks good. One minor code style note, and one place where you can remove some code that is redundant given Django's default behavior.

Commit:
LIBITD-2552. Implemented changes suggested

Renamed group list file to use snake case and removed
query set function in the GroupList view

https://umd-dit.atlassian.net/browse/LIBITD-2552

Commit:
LIBITD-2552. Updated template name in GroupListView

https://umd-dit.atlassian.net/browse/LIBITD-2552

peichman-umd commented on src/ipmanager/ui/templates/ui/group_list_view.html:
```diff
@@ -0,0 +1,36 @@
+{% block content %}
+
+
+<h1>IPManager</h1>
```
I would suggest removing this heading, since it is redundant with the application name in the page header. You can also add a `title` value to the context and the django-umd-lib-style base template will add it as an `h1` header in the body.

```python
class GroupListView(ListView):
    ...
    def get_context_data(self, **kwargs):
        context = self.get_context_data(**kwargs)
        context['title'] = 'All Groups'
        return context
```

peichman-umd reviewed (changes requested):
One more change regarding the page heading and title.

Commit:
LIBITD-2552. Removed Header from template

https://umd-dit.atlassian.net/browse/LIBITD-2552
